src/main/java/seedu/address/model/person/Analytics.java

```diff
@@ -0,0 +1,227 @@
+        }
+        if (this.numLoans > 0) {
+            this.propOverdueLoans = (float) this.numOverdueLoans / this.numLoans;
+        }
+    }
+
+    /**
+     * Updates the fields that calculate the total value of various loans.
+     * @param loan The loan to update the fields with.
+     */
+    private void updateValueFields(Loan loan) {
+        this.totalValueLoaned += loan.getValue();
+        if (loan.isOverdue()) {
+            this.totalValueOverdue += loan.getValue();
+        }
+        if (loan.isActive()) {
+            this.totalValueActive += loan.getValue();
+        }
+    }
+
+    /**
+     * Updates the fields that calculate the average value of various loans.
+     * This method should be called after the fields that calculate the total value of various loans have been updated.
+     */
+    private void updateAverageFields() {
+        if (this.numActiveLoans > 0) {
+            this.averageActiveValue = this.totalValueActive / this.numActiveLoans;
+        }
+        if (this.numOverdueLoans > 0) {
+            this.averageOverdueValue = this.totalValueOverdue / this.numOverdueLoans;
+        }
+        if (this.numLoans > 0) {
+            this.averageLoanValue = this.totalValueLoaned / this.numLoans;
+        }
+    }
+
+    /**
+     * Updates the fields that calculate the earliest and latest dates of various loans.
+     * @param loan The loan to update the fields with.
+     */
+    private void updateDateFields(Loan loan) {
+        if (this.earliestLoanDate == null || loan.getStartDate().before(this.earliestLoanDate)) {
+            this.earliestLoanDate = loan.getStartDate();
+        }
+        if (this.latestLoanDate == null || loan.getStartDate().after(this.latestLoanDate)) {
+            this.latestLoanDate = loan.getStartDate();
+        }
+        if (!loan.isReturned()) {
+            if (this.earliestReturnDate == null || loan.getReturnDate().before(this.earliestReturnDate)) {
+                this.earliestReturnDate = loan.getReturnDate();
+            }
+            if (this.latestReturnDate == null || loan.getReturnDate().after(this.latestReturnDate)) {
+                this.latestReturnDate = loan.getReturnDate();
+            }
+        }
+    }
+
+    /**
+     * Returns an Analytics object that represents the analytics of a LoanRecords object.
+     * @param loanRecords The LoanRecords object to get the analytics from.
+     * @return The Analytics object that represents the analytics of the LoanRecords object.
+     */
+    public static Analytics getAnalytics(LoanRecords loanRecords) {
+        Analytics analytics = new Analytics();
+        for (int i = 0; i < loanRecords.size(); i++) {
+            Loan loan = loanRecords.getLoan(i);
+            analytics.updateNumFields(loan);
+            analytics.updateValueFields(loan);
+            analytics.updateDateFields(loan);
+        }
+        analytics.updatePropFields();
+        analytics.updateAverageFields();
```

Comment:
It seems that the update methods must be called in a specific order to prevent errors. While it is not a concern now because each `Analytics` object is effectively immutable and `getAnalytics()` is the only way of obtaining it, it would be good if this can be addressed so that future iterations do not run into related bugs.